internal/cloudapi/v1/v1.go

```diff
@@ -157,25 +158,45 @@ func (h *apiHandlers) Compose(ctx echo.Context) error {
 		}
 
 		packageSets := imageType.PackageSets(bp)
-		pkgSpecSets := make(map[string][]rpmmd.PackageSpec)
-		for name, packages := range packageSets {
-			pkgs, _, err := h.server.rpmMetadata.Depsolve(packages, repositories, distribution.ModulePlatformID(), arch.Name(), distribution.Releasever())
+		depsolveJobID, err := h.server.workers.EnqueueDepsolve(&worker.DepsolveJob{
+			PackageSets:      packageSets,
+			Repos:            repositories,
+			ModulePlatformID: distribution.ModulePlatformID(),
+			Arch:             arch.Name(),
+			Releasever:       distribution.Releasever(),
+		})
+		if err != nil {
+			panic(err)
+		}
+
+		var depsolveResults worker.DepsolveJobResult
+		for {
+			status, _, err := h.server.workers.JobStatus(depsolveJobID, &depsolveResults)
 			if err != nil {
-				var error_type int
-				switch err.(type) {
-				// Known DNF errors falls under BadRequest
-				case *rpmmd.DNFError:
-					error_type = http.StatusBadRequest
-				// All other kind of errors are internal server Errors.
-				// (json marshalling issues for instance)
-				case error:
-					error_type = http.StatusInternalServerError
-				}
-				return echo.NewHTTPError(error_type, "Failed to depsolve base packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err)
+				panic(err)
+			}
+			if status.Canceled {
+				panic("unexpected cancellation of a depsolve job")
+			}
+			if !status.Finished.IsZero() {
+				break
 			}
-			pkgSpecSets[name] = pkgs
+			time.Sleep(500 * time.Millisecond)
```

Comment:
Considering depsolving should take ~100ms once we no longer flush the caches, maybe  make the sleep 50ms rather than 500?

Reply:
yep, that's a good point